.github/workflows/ci-linux.yml

```diff
@@ -77,13 +77,26 @@ jobs:
         run: |
           export cache_dir="${{ env.CACHE_DIR }}"
           export CCACHE_COMPILERCHECK="string:$(clang --version)"
-          bash build_tools/ci/build_test_cpp.sh
+          bash build_tools/build_llvm.sh
+          rm -rf llvm-build
+          export llvm_install_dir=$PWD/llvm-install
+          bash build_tools/build_test_cpp.sh
 
       - name: Create artifacts
         if: ${{ !cancelled() }}
         run: |
+          pushd third_party/iree/third_party/llvm-project && llvm_sha_short=$(git rev-parse --short HEAD) && popd
```

Comment:
Nice one liner for getting the llvm sha. I've been considering using the llvm sha in cache keys, since that represents what _mostly_ affects build time. Getting a cache hit from the same LLVM commit is probably more useful than a cache hit from a more recent timestamp, and cache uploads could be limited to only one per llvm sha.

Reply:
i've considered something like that before but i think it's not worth it - like how often do you have multiple shas of llvm being "asked for" by various PRs that would avail themselves of a cache hit? with how small the free cache limit is (10Gb) i think the answer is basically never?